scio-bigtable/src/main/scala/com/spotify/scio/bigtable/TableAdmin.scala

```diff
@@ -112,34 +158,50 @@ object TableAdmin {
   private def ensureColumnFamilies(
     client: BigtableTableAdminClient,
     tablePath: String,
-    columnFamilies: List[String]
+    columnFamilies: List[(String, Option[Duration])]
   ): Unit = {
-
     val tableInfo =
       client.getTable(GetTableRequest.newBuilder().setName(tablePath).build)
 
-    val modifications: List[Modification] = columnFamilies.collect {
-      case (cf) if !tableInfo.containsColumnFamilies(cf) =>
-        Modification
-          .newBuilder()
-          .setId(cf)
-          .setCreate(ColumnFamily.newBuilder())
-          .build()
-    }
+    val cfList = columnFamilies
+      .map {
+        case (n, duration) =>
+          val gcRule = duration.map(gcRuleFromDuration).getOrElse(GcRule.getDefaultInstance())
+          val cf = tableInfo
+            .getColumnFamiliesOrDefault(n, ColumnFamily.newBuilder().build())
+            .toBuilder()
+            .setGcRule(gcRule)
+            .build()
 
-    if (modifications.isEmpty) {
-      log.info(s"Column families $columnFamilies exist in $tablePath")
-    } else {
-      log.info(s"Creating column families $columnFamilies in $tablePath")
-      client.modifyColumnFamily(
-        ModifyColumnFamiliesRequest
-          .newBuilder()
-          .setName(tablePath)
-          .addAllModifications(modifications.asJava)
-          .build
-      )
-      ()
-    }
+          (n, cf)
+      }
+
+    val modifications = cfList
+      .map {
+        case (n, cf) =>
+          val mod = Modification.newBuilder().setId(n)
+          if (tableInfo.containsColumnFamilies(n)) {
+            mod.setUpdate(cf)
+          } else {
+            mod.setCreate(cf)
+          }
+
+          mod.build()
+      }
+
```

Comment:
nit: maybe a log line here? Something like:

```suggestion
    log.info("Modifying or updating {} column families for table {}", modifications.size, tablePath)
```